great_expectations/data_context/store/datasource_store.py

```diff
@@ -215,6 +215,16 @@ def _build_key_from_config(  # type: ignore[override]
         )
         return self.store_backend.build_key(name=datasource_config.name, id=id_)
 
+    def get_fluent_datasource_by_name(self, name: str) -> FluentDatasource:
```

Comment:
I got sick of writing `isinstance(datasource, Datasource)` everywhere, so this at least removes some of the need for that.